src/software/ai/passing/cost_function.cpp

```diff
@@ -8,70 +8,66 @@
 #include "software/ai/evaluation/pass.h"
 #include "software/geom/algorithms/acute_angle.h"
 #include "software/geom/algorithms/closest_point.h"
+#include "software/geom/algorithms/contains.h"
 #include "software/logger/logger.h"
 
-double ratePass(const World& world, const Pass& pass,
-                const std::optional<Rectangle>& target_region,
-                std::optional<unsigned int> passer_robot_id, PassType pass_type,
```

Comment:
Just curious, does it not get a bit difficult to optimize for a cross-field pass (ex zone 7->9)? I'd imagine the shoot score drags all the values down and you get a really small gradient to follow. I would imagine that you'd want to choose a pass-and-dribble for some pairs of zones, and a one-touch pass for others? I'm sure I'm not right here having not actually played with this, but curious if this was an issue / consideration.

Reply:
I'd be super interested in seeing a visualization of pass-score across the field for these cases, may help build some useful intuition.

Reply:
Yeah it does get pretty difficult mostly because the rotation required to do a one-touch is >> 60 degrees for a pass from zone 7 -> 9 (the sigmoid responsible pulls the score down to 0) 

A temporary solution I found was to say that any passes w/ the passer or receiver location on the friendly side can rotate 180 degrees to take a shot on net, and any passes on the enemy can only rotate up to 60 degrees (default one-touch we have now)

This generally fixes the problem, but def need a more robust long term solution which will be a part of #1987